test/e2e/proxy_publicviewer_test.go

```diff
@@ -0,0 +1,225 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	waitpoll "k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	commonauth "github.com/codeready-toolchain/toolchain-common/pkg/test/auth"
+	testconfig "github.com/codeready-toolchain/toolchain-common/pkg/test/config"
+	testspace "github.com/codeready-toolchain/toolchain-common/pkg/test/space"
+	. "github.com/codeready-toolchain/toolchain-e2e/testsupport"
+	authsupport "github.com/codeready-toolchain/toolchain-e2e/testsupport/auth"
+	testsupportspace "github.com/codeready-toolchain/toolchain-e2e/testsupport/space"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport/spacebinding"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport/tiers"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport/wait"
+)
+
+type proxyUser struct {
+	expectedMemberCluster *wait.MemberAwaitility
+	username              string
+	token                 string
+	identityID            uuid.UUID
+	signup                *toolchainv1alpha1.UserSignup
+	compliantUsername     string
+}
+
+// tests access to community-shared spaces
+func TestProxyPublicViewer(t *testing.T) {
+	// given
+
+	// make sure everything is ready before running the actual tests
+	awaitilities := WaitForDeployments(t)
+	hostAwait := awaitilities.Host()
+	memberAwait := awaitilities.Member1()
+
+	// public viewer is enabled in ToolchainConfig
+	hostAwait.UpdateToolchainConfig(t, testconfig.PublicViewerConfig(true))
+
+	// we create a space to share and a user for 'community-user'
+	space, _, _ := testsupportspace.CreateSpace(t, awaitilities, testspace.WithTierName("appstudio"), testspace.WithSpecTargetCluster(memberAwait.ClusterName))
+	communityUser := &proxyUser{
+		expectedMemberCluster: memberAwait,
+		username:              "community-user",
+		identityID:            uuid.New(),
+	}
+	createAppStudioUser(t, awaitilities, communityUser)
+	communityUserProxyClient, err := hostAwait.CreateAPIProxyClient(t, communityUser.token, hostAwait.APIProxyURL)
+	require.NoError(t, err)
+
+	t.Run("space is flagged as community", func(t *testing.T) {
+		// when
+		sb := CreateCommunitySpaceBinding(t, hostAwait, space.Name, space.Namespace)
+		require.NotNil(t, sb)
+
+		// Wait until space is flagged as community
+		require.NoError(t,
+			waitpoll.Poll(hostAwait.RetryInterval, hostAwait.Timeout, func() (bool, error) {
+				opts := []client.ListOption{
+					client.MatchingLabels{
+						toolchainv1alpha1.SpaceBindingMasterUserRecordLabelKey: toolchainv1alpha1.KubesawAuthenticatedUsername,
+						toolchainv1alpha1.SpaceBindingSpaceLabelKey:            space.Name,
+					},
+					client.InNamespace(space.Namespace),
+				}
+				sbs := &toolchainv1alpha1.SpaceBindingList{}
+				if err := hostAwait.Client.List(context.TODO(), sbs, opts...); err != nil {
+					return false, err
+				}
+
+				return len(sbs.Items) == 1, nil
+			}))
```

Comment:
you could use
```suggestion
				require.NoError(t, err)
				_, err = awaitilities.Host().WaitForSpaceBinding(t, toolchainv1alpha1.KubesawAuthenticatedUsername, space.Name,
					UntilSpaceBindingHasSpaceRole("viewer"))
				require.NoError(t, err)
```
this way you also check the expected role.

Reply:
done in 8d2872804d812e52a62d4144e4451b21db20e798